editoast/src/views/infra/delimited_area.rs

```diff
@@ -0,0 +1,837 @@
+                };
+                let neighbours = graph
+                    .get_all_neighbours(&opposite_track_endpoint)
+                    .into_iter()
+                    .map(|neighbour| (neighbour, neighbours_remaining_distance))
+                    .collect::<Vec<_>>();
+                next_tracks.extend(neighbours);
+            }
+        }
+    }
+    related_track_ranges
+}
+
+/// Return the closest exit that applies on a track from a starting endpoint.
+/// To be applicable, an exit must be in the correct direction.
+fn closest_exit_from_endpoint<'a>(
+    track_endpoint: &TrackEndpoint,
+    exits: Option<&'a Vec<&DirectedLocation>>,
+) -> Option<&'a DirectedLocation> {
+    if let Some(exits) = exits {
+        exits
+            .iter()
+            .filter(|exit| exit.track == track_endpoint.track)
+            .filter(|exit| match track_endpoint.endpoint {
+                Endpoint::Begin => exit.direction == Direction::StartToStop,
+                Endpoint::End => exit.direction == Direction::StopToStart,
+            })
+            .sorted_by(
+                |e_1, e_2| match (track_endpoint.endpoint, e_1.position < e_2.position) {
+                    (Endpoint::Begin, true) | (Endpoint::End, false) => Ordering::Less,
+                    (Endpoint::Begin, false) | (Endpoint::End, true) => Ordering::Greater,
+                },
+            )
+            .map(|exit| &**exit)
+            .next()
+    } else {
+        None
+    }
+}
+
+/// Return the closest applicable exit that is on the same track as the `entry`, or `None`.
+/// if there is none.
+fn closest_exit_from_entry<'a>(
+    entry: &DirectedLocation,
+    exits: Option<&'a Vec<&DirectedLocation>>,
+) -> Option<&'a DirectedLocation> {
+    if let Some(exits) = exits {
+        exits
+            .iter()
+            .filter(|exit| exit.track == entry.track)
+            .filter(|exit| entry.direction == exit.direction)
+            .filter(|exit| match entry.direction {
+                Direction::StartToStop => entry.position < exit.position,
+                Direction::StopToStart => exit.position < entry.position,
+            })
+            .sorted_by(
+                |e_1, e_2| match (entry.direction, e_1.position < e_2.position) {
+                    (Direction::StartToStop, true) | (Direction::StopToStart, false) => {
+                        Ordering::Less
+                    }
+                    (Direction::StartToStop, false) | (Direction::StopToStart, true) => {
+                        Ordering::Greater
+                    }
+                },
+            )
+            .map(|exit| &**exit)
+            .next()
+    } else {
+        None
+    }
+}
+
+/// Return the directional track range starting at `entry` finishing at `exit`, or `None`
+/// if no track range can be built from them.
+fn track_range_between_two_locations(
+    entry: &DirectedLocation,
+    exit: &DirectedLocation,
+) -> Option<DirectionalTrackRange> {
```

Comment:
```suggestion
) -> Result<DirectionalTrackRange> {
```

Reply:
This signature got affected by the changes mentioned in https://github.com/OpenRailAssociation/osrd/pull/9381#discussion_r1848511107. I updated the signature to use `std::result::Result` instead of `editoast_error::Result` since these are internal errors that will cause panics and never return to the endpoint handler. What do you think ?

Reply:
Looks to me like the best thing to do here. Since `EditoastError` will be deprecated "soon", let's avoid using it in places where we can do something else.